### PR TITLE
[analytics] Send URL in published events

### PR DESCRIPTION
**Pull request by caropinzonsilva**
## 🎩 How can this be tophatted?

...

## ✅ Checklist

- [ ] Tested on mobile
- [ ] Tested on multiple browsers
- [ ] Tested for accessibility
- [ ] Includes unit tests
- [ ] Updated relevant documentation for the changes (if necessary)

### CODE REVIEW

File: packages/blockchain-components/src/utils/analytics/ClientAnalytics/utils.test.tsx [hunk 1]
```diff
@@ -18,8 +18,9 @@ describe('utils', () => {
 
   beforeEach(() => {
     vi.useFakeTimers();
-    // We only need to mock the pathname for these tests
-    (window as {location: {pathname: string}}).location = {
+    // We only need to mock the href and pathname for these tests
```
[the remark this answers is not in the document]
Comment (caropinzonsilva) [reply packages/blockchain-components/src/utils/analytics/ClientAnalytics/utils.test.tsx:21]: This edge case is already covered in a later test: https://github.com/Shopify/blockchain-components/pull/79/files#diff-f538bbf837d085a2ea71f2fa715997e6eff2522da31d3b712fbbbd0a6238b265R283
